src/drive/web/modules/actions/useActions.spec.jsx

```diff
@@ -0,0 +1,422 @@
+import React from 'react'
+import { renderHook } from '@testing-library/react-hooks'
+import { createMockClient } from 'cozy-client'
+
+import { ModalContext } from 'drive/lib/ModalContext'
+import { SharingContext, ShareModal } from 'cozy-sharing'
+import { RouterContext } from 'drive/lib/RouterContext'
+import AppLike from '../../../../../test/components/AppLike'
+import DeleteConfirm from 'drive/web/modules/drive/DeleteConfirm'
+import MoveModal from 'drive/web/modules/move/MoveModal'
+import { EditDocumentQualification } from 'cozy-scanner'
+import { createStore } from 'redux'
+import { exportFilesNative, downloadFiles, openFileWith } from './utils'
+import * as rename from 'drive/web/modules/drive/rename'
+
+import useActions from './useActions'
+
+jest.mock('./utils', () => ({
+  isAnyFileReferencedByAlbum: jest.fn(),
+  exportFilesNative: jest.fn(),
+  downloadFiles: jest.fn(),
+  trashFiles: jest.fn(),
+  openFileWith: jest.fn()
+}))
+
+describe('useActions', () => {
+  const mockFolderId = '123'
+  const mockStore = createStore(() => ({
+    mobile: {
+      url: 'cozy-url://'
+    }
+  }))
+  const mockModalContextValue = {
+    pushModal: jest.fn()
+  }
+  const mockSharingContextValue = {
+    hasWriteAccess: jest.fn(),
+    refresh: jest.fn()
+  }
+  const mockRouterContextValue = {
+    router: {
+      push: jest.fn()
+    },
+    location: { pathname: '/folder' }
+  }
+  const mockClient = createMockClient({})
+
+  beforeEach(() => {
+    jest.resetAllMocks()
+    global.__TARGET__ = 'desktop'
+  })
+
+  const renderActionsHook = (...hookParams) => {
+    const wrapper = ({ children }) => (
+      <AppLike client={mockClient} store={mockStore}>
+        <ModalContext.Provider value={mockModalContextValue}>
+          <SharingContext.Provider value={mockSharingContextValue}>
+            <RouterContext.Provider value={mockRouterContextValue}>
+              {children}
+            </RouterContext.Provider>
+          </SharingContext.Provider>
+        </ModalContext.Provider>
+      </AppLike>
+    )
+    return renderHook(() => useActions(...hookParams), {
+      wrapper
+    })
+  }
+
+  const defaultHookArgs = [
+    mockFolderId,
+    {
+      canMove: true
+    }
+  ]
+  const getAction = (actionKey, hookArgs = defaultHookArgs) => {
+    const { result } = renderActionsHook(...hookArgs)
+    return result.current[actionKey]
+  }
+
+  it('returns actions keyed by icon', () => {
```

Comment:
If it is a good moment to have the actions keyed by id, I think we should do it now and repeat potentially the icon in the action spec. I do not think having the icon as key is a good idea.

Reply:
I think actions should be keyed by id and not icon, in fact in the definition they are keyed by id. But the component in cozy-ui expects icons names as keys and a `displayCondition` function. I'll create an issue in cozy-ui about that.

In the meantime, I think it makes sense to keep the name `displayCondition` for "grep-ability", and for the keys… we could change the key further down the line, but we'd have to do it in multiple places, so I suspect that would end up being more confusing.